src/common/src/types/decimal.rs

```diff
@@ -31,6 +33,17 @@ pub enum Decimal {
     NegativeInf,
 }
 
+impl ToText for crate::types::Decimal {
+    fn to_text(&self) -> String {
+        match self {
```

Comment:
Why not just call `self.to_string()`? (And same for interval)

Reply:
This can separate them if we want to remove to_string in the future.

Reply:
By that time we can just move code from `Display` into `ToText`. The interface of `ToText` remains the same and no updates to callers are needed. Keeping two duplicated snippets would be hard to maintain.